modyn/selector/selector.py

```diff
@@ -96,10 +96,11 @@ def get_number_of_samples(self, trigger_id: int) -> int:
 
     def get_status_bar_scale(self) -> int:
         # the status bar scale is only meaningful if we are using a Downsampling strategy. Otherwise, it's always 100%
-        if not isinstance(self._strategy, AbstractDownsampleStrategy):
+        if not isinstance(self._strategy, CoresetStrategy):
             return 100
 
-        return self._strategy.get_training_status_bar_scale()
+        self._strategy.inform_next_trigger()
```

Comment:
Hm, so this is still not really solving the core problem that I mentioned: A getter should not have any side effects. 

The get_status_bar_scale function should do exactly that: Return the status bar scale. It should not matter if I ask it once, twice, thrice, etc. However, right now, every time we call this function, the CoresetScheduler makes one step forward. This should not happen. A getter should not change some properties. It might do calculations, but advancing something is not ok. Especially here, since we introduce the assumption that get_status_bar_scale is called with a certain pattern, and that assumption is hardcoded into the selector currently.

Reply:
I partially disagree with your comment. The function is idempotent, you can call it 1000 times in the same trigger and nothing changes. But I agree that at the code level it is not exceptional. I removed it from there and put it in `trigger()`

Reply:
Hmm, as far as I was able to trace it, `inform_next_trigger` got forwarded up to the scheduler, which increased it sampler by one. However, putting it in `on_trigger` is cleaner in any case.